src/hotspot/share/asm/codeBuffer.hpp

```diff
@@ -748,6 +749,12 @@ inline bool CodeSection::maybe_expand_to_ensure_remaining(csize_t amount) {
 
 inline int CodeSection::alignment(int section) {
   if (section == CodeBuffer::SECT_CONSTS) {
+#if INCLUDE_JVMCI
+    if (EnableJVMCI) {
+      // Graal vectorization requires larger aligned constants
+      return 64;
```

Comment:
This means all Graal installed code will pay a penalty even though most installed code does not include constants that need such large alignment. It would be preferable to allow a compiler to specify the alignment requirement per nmethod.

Reply:
I agree. The alignment should probably be a field in CodeBuffer with a default value of 8.